src/sec_edgar_bulker/core/utils.py

```
@@ -0,0 +1,168 @@
+            if parsed.year != test_datetime.year:
+                raise ValueError("Format string must include year")
+        except ValueError as e:
+            raise ValueError(f"Invalid datetime format string: {str(e)}")
+            
+        # Generate actual batch ID
+        now = datetime.now()
+        batch_id = now.strftime(format_str)
+        
+        # Validate the generated ID
+        if not batch_id or len(batch_id) < 8:  # Minimum length for a meaningful ID
+            raise ValueError("Generated batch ID is too short")
```

Ruff (0.8.2)

Abstract `raise` to an inner function

(TRY301)